processor/samplingprocessor/probabilisticsamplerprocessor/probabilisticsampler.go

```diff
@@ -69,10 +74,21 @@ func (tsp *tracesamplerprocessor) ConsumeTraceData(ctx context.Context, td consu
 
 	sampledSpans := make([]*tracepb.Span, 0, len(td.Spans))
 	for _, span := range td.Spans {
-		// If one assumes random trace ids hashing may seems avoidable, however, traces can be coming from sources
-		// with various different criteria to generate trace id and perhaps were already sampled without hashing.
-		// Hashing here prevents bias due to such systems.
-		if hash(span.TraceId, tsp.hashSeed)&bitMaskHashBuckets < scaledSamplingRate {
+		samplingPriority := parseSpanSamplingPriority(span)
+		if samplingPriority == doNotSampleSpan {
+			// Take a restrictive approach since some may use this to remove spans from traces.
+			continue
+		}
+
+		sampled :=
+			// First check sampling priority per OpenTracing semantics.
+			samplingPriority == mustSampleSpan ||
+				// If one assumes random trace ids hashing may seems avoidable, however, traces can be coming from sources
```

Comment:
This indentation is put here by gofmt

Reply:
Nit: split the expression and use temp variables to help gofmt behave. It would also likely make it more readable.